src/main/java/hudson/matrix/MatrixAggregatable.java

```diff
@@ -33,7 +34,7 @@
 /**
  * {@link Publisher}, {@link JobProperty}, {@link BuildWrapper} can optionally implement this interface
  * to perform result aggregation across {@link MatrixRun}.
- *
+ * <p>You may also register singleton instances of this as {@link Extension}s.
```

Comment:
This is a great improvement but I am not sure I would understand this from the comment.

If I am not mistaken, these would actually be "global" aggregatos as they will not be configured in job (unlike Publisher, JobProperty, or BuildWrapper). I find this worth mentioning.